File: demo_v2_threshold/demo_v2/backend/app/schemas/excel.py
"""Excel import API schemas."""

from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, Field


class ExcelImportRowPreview(BaseModel):
    id: UUID
    metric_code: str
    raw_value: str | None
    parsed_value: Decimal | None
    validation_errors: list[str]
    row_number: int


class ExcelImportBatchResponse(BaseModel):
    id: UUID
    filename: str
    status: str
    submission_id: UUID | None
    uploaded_at: datetime
    validation_summary: dict | None
    rows: list[ExcelImportRowPreview]


class ExcelApplyRowEdit(BaseModel):
    metric_code: str
    value: Decimal | float | int | str


class ExcelApplyRequest(BaseModel):
    submission_id: UUID
    rows: list[ExcelApplyRowEdit] | None = Field(
        default=None,
        description="PM-edited values; omit to use parsed values from batch",
    )

File: demo_v2_threshold/demo_v2/frontend/src/types/excel.ts
export interface ExcelImportRowPreview {
  id: string;
  metric_code: string;
  raw_value: string | null;
  parsed_value: number | null;
  validation_errors: string[];
  row_number: number;
}

export interface ExcelImportBatch {
  id: string;
  filename: string;
  status: string;
  submission_id: string | null;
  uploaded_at: string;
  validation_summary: {
    total_rows: number;
    valid_rows: number;
    invalid_rows: number;
    error_count: number;
  } | null;
  rows: ExcelImportRowPreview[];
}

export interface ExcelApplyRowEdit {
  metric_code: string;
  value: number | string;
}

export interface ExcelApplyRequest {
  submission_id: string;
  rows: ExcelApplyRowEdit[] | null;
}
